### PR TITLE
Fix CLI to avoid deprecation warning

### PR DESCRIPTION
**Pull request by YakDriver**

### CODE REVIEW

**Comment by YakDriver** [comment]
After the new CLI check, the `kwargs` dict will have all 5 keys always. If they weren't in `kwargs` prior to the check, they'll have the value of `None` (Pythonic). Also, any that have the value of `'None'` (string, case insensitive), will also have the value of `None` (Pythonic).

**Comment by YakDriver** [comment]
@lorengordon What do you think of this approach? This way it only cleans up None strings when specifically requested. Thus, YAML-source attributes will still result in deprecation warnings but CLI won't. Also, I added some simple tests.

**Comment by lorengordon** [comment]
The config-sourced arguments are handled in the `Watchmaker()` class, so no change to `Arguments()` will impact them either way.

I also do not think we should hard code the list of noneables, since the each of these arguments are handled individually already in the class. For example, we can just do:

```
        self.admin_groups = _clean_none(kwargs.pop('admin_groups', None))
```

We probably also want to handle `extra_arguments`...

```
        self.extra_arguments = [
            _clean_none(val) for val in kwargs.pop('extra_arguments', None) or []
        ]
```

**Comment by lorengordon** [comment]
Also, if `clean_none` were a utils function, there's probably a reasonably simple way to have the `config_none_deprecate` use it to avoid repeating the logic.

**Comment by YakDriver** [comment]
I didn't realize Arguments was only for the CLI. I've made the changes. Let me know if there's anything else.

File: src/watchmaker/utils/__init__.py [hunk 1]
```diff
@@ -101,12 +101,20 @@ def config_none_deprecate(check_value, log):
             Logger where deprecation warning will be made.
 
     """
-    if check_value and check_value.lower() == 'none':
+    # only warn when 1 - from YAML, 2 - type string, 3 - "none"
+    if isinstance(check_value, str) and check_value.lower() == 'none':
```
[the remark this answers is not in the document]
Comment (YakDriver) [reply src/watchmaker/utils/__init__.py]: This is fine with me.

**Pull request merged**